### PR TITLE
Exclude custom data directory from integrity checker

### LINKED ISSUE #22698: oc 9.0 beta2: file integrity check
### Steps to reproduce
1. made a copy of small production oc 8.2.2
2. rename of old oc, upload 9.0 beta2, moved config.php and data-folder to new oc
3. perform web-update

### Expected behaviour
file integrity: ok

### Actual behaviour
file integrity: reports error, EXTRA_FILE

### Server configuration
**Operating system:** Shared webhost: Linux dd27002 3.2.0-98-generic #138-Ubuntu SMP Mon Jan 11 12:33:01 UTC 2016 x86_64

**Web server:** Apache

**Database:** MySQL

**PHP version:** 5.5.31

**ownCloud version:** ownCloud 9.0.0 beta 2 (testing)

**Updated from an older ownCloud or fresh install:** update from 8.2.2

**Signing status (ownCloud 9.0 and above):** 
- core
	- EXTRA_FILE
		- data_ufdruqd6b2/admin/thumbnails/7/1024-768-max.png
		- data_ufdruqd6b2/admin/thumbnails/7/200-200.png
		- data_ufdruqd6b2/admin/thumbnails/7/267-200-with-aspect.png
		- data_ufdruqd6b2/admin/thumbnails/9/300-200-with-aspect.png
		- data_ufdruqd6b2/admin/thumbnails/9/1024-683-max.png
		- data_ufdruqd6b2/admin/thumbnails/9/200-200.png
		- data_ufdruqd6b2/admin/thumbnails/8/300-200-with-aspect.png
		- data_ufdruqd6b2/admin/thumbnails/8/1024-683-max.png
		- data_ufdruqd6b2/admin/thumbnails/8/200-200.png
		- data_ufdruqd6b2/admin/files/ownCloud_User_Manual.pdf
		- data_ufdruqd6b2/owncloud.log
		- data_ufdruqd6b2/updater_backup/8.2.1.4-9f264f98.zip
		- data_ufdruqd6b2/updater_backup/8.2.1.4.zip
		- data_ufdruqd6b2/updater_backup/8.2.2.2.zip
		- data_ufdruqd6b2/files_external/rootcerts.crt
		- data_ufdruqd6b2/.htaccess

**Are you using external storage, if yes which one:** no

**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** no

### PR DESCRIPTION
**Pull request by LukasReschke**
We should not scan any custom data directory in the integrity checker as well. Otherwise this would lead to a massive increased update time that may likely exceed the timeout.

To test this:
1. Install ownCloud 8.2.2, set as data dir another folder than `data` such as `data-asdf`. Make sure that folder lives in the web root as well.
2. Update that instance to 9.0.0 beta2 => Errors are shown in the code integrity check
3. Apply this patch and trigger the rescan in the admin settings => Only one errors is shown and this is this file (can also do a new update with that patch applied)

Fixes https://github.com/owncloud/core/issues/22698

### CODE REVIEW

File: lib/private/integritycheck/iterator/excludefoldersbypathfilteriterator.php [hunk 1]
```diff
@@ -32,12 +32,18 @@ public function __construct(\RecursiveIterator $iterator, $root = '') {
...
 
-		$this->excludedFolders = array_merge([
+		$excludedFolders = [
 			rtrim($root . '/data', '/'),
 			rtrim($root .'/themes', '/'),
 			rtrim($root.'/config', '/'),
 			rtrim($root.'/apps', '/'),
-		], $appFolders);
+		];
+		$customDataDir = \OC::$server->getConfig()->getSystemValue('datadirectory', '');
```
Comment (LukasReschke) [inline lib/private/integritycheck/iterator/excludefoldersbypathfilteriterator.php]: Injecting into iterators is no fun. Thus directly the access, we're anyways doing similar stuff above where we access  `\OC::$APPSROOTS` directly.

**Comment by PVince81** [comment]
:+1: makes sense

**Comment by rullzer** [comment]
People should not have their data directory in the root.. :wink: 
Works properly over here. :+1:

**Comment by PVince81** [comment]
CI is stuck for integration tests.
I ran them locally just in case and they look fine.

This can be merged.

**Comment by karlitschek** [comment]
makes sense :+1:

**Comment by akhokhlushin** [comment]
Standard gentoo tool for managing webapps (webapp-config) makes in every webapp root dir two files contains base webapp info and hashes. These two files needed for webapp automatic update  process.
If I delete this files, webapp-config will not work, if i don't delete - owncloud will not work.

I think people have to have an opportunity to exclude files from check

**Comment by RealRancor** [comment]
@akhokhlushin Please create a new feature request for the exclusion of specific files.

**Pull request merged**